test.py

```diff
@@ -0,0 +1,247 @@
+            'unzip', '-qq', filePath,
+            '-d', dest
+        ])
+
+def BuildAddOn (rootFolder, buildFolder, devKitFolder, addOnName, platformName, configuration, optionalParams, languageCode=None):
+    buildPath = buildFolder / addOnName
+    if languageCode is not None:
+        buildPath = buildPath / languageCode
+
+    # Add params to configure cmake
+    projGenParams = []
+    projGenParams.append ('cmake')
+    projGenParams.extend (['-B', str(buildPath)])
+
+    if platformName == 'WIN':
+        projGenParams.extend (['-G', 'Visual Studio 16 2019'])
+    elif platformName == 'MAC':
+        projGenParams.extend (['-G', 'Xcode'])
+
+    projGenParams.append ('-DAC_API_DEVKIT_DIR=' + str(devKitFolder / 'Support'))
+    projGenParams.append ('-DCMAKE_BUILD_TYPE=' + configuration)
+
+    if languageCode is not None:
+        projGenParams.append ('-DAC_ADDON_LANGUAGE=' + languageCode)
+
+    if optionalParams is not None:
+        for key in optionalParams:
+            projGenParams.append ('-D' + key + '=' + optionalParams[key])
+
+    projGenParams.append (str(rootFolder))
+
+    projGenResult = subprocess.call (projGenParams)
+    if projGenResult != 0:
+        print ('Failed to generate project')
+        return 1
+    
+    # Add params to build AddOn
+    buildParams = []
+    buildParams.append ('cmake')
+    buildParams.extend (['--build', str(buildPath)])
+    buildParams.extend (['--config', configuration])
+
+    buildResult = subprocess.call (buildParams)
+    if buildResult != 0:
+        print ('Failed to build project')
+        return 1
+    
+    return 0
+
+def CopyResultToPackage(packageRootFolder, buildFolder, addOnName, platformName, configuration, languageCode=None, isRelease=False):
+    packageFolder = packageRootFolder
+    sourceFolder = buildFolder / addOnName
+
+    if languageCode is not None:
+        packageFolder = packageFolder / languageCode
+        sourceFolder = sourceFolder / languageCode
+    sourceFolder = sourceFolder / configuration
+
+    if not packageFolder.exists():
+        packageFolder.mkdir(parents=True)
+
+    fileName = addOnName
+    if not isRelease:
+        fileName = fileName + '_' + configuration
+
+    if platformName == 'WIN':
+        shutil.copy (
+            sourceFolder / (addOnName + '.apx'),
+            packageFolder / (fileName + '.apx')
+        )
+        if configuration == 'Debug':
+            shutil.copy (
+                sourceFolder / (addOnName + '.pdb'),
+                packageFolder / (fileName + '.pdb')
+            )
+
+    elif platformName == 'MAC':
+        subprocess.call ([
+            'cp', '-r',
+            sourceFolder / (addOnName + '.bundle'),
+            packageFolder / (fileName + '.bundle')
+        ])
+    
+
+
+def Main():
+    parser = argparse.ArgumentParser ()
+    parser.add_argument ('--acVersion', dest = 'acVersion', type = str, required = True, help = 'Archicad version number list. Ex: "26 27"')

```

Comment:
Argument parsing should be in a separate function. Main should only contain the line: args = ParseArguments()

Reply:
Moved some logic out of the Main function, but the initial verifications and variable assignments are a bit tricky to separate